clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperation.java

```diff
@@ -337,14 +337,17 @@ protected RegisterTaskDefinitionRequest makeTaskDefinitionRequest(
   protected RegisterTaskDefinitionRequest makeTaskDefinitionRequestFromArtifact(
       String ecsServiceRole, EcsServerGroupName newServerGroupName) {
 
-    File artifactFile =
-        downloadTaskDefinitionArtifact(description.getResolvedTaskDefinitionArtifact());
-
     RegisterTaskDefinitionRequest requestTemplate;
-    try {
-      requestTemplate = mapper.readValue(artifactFile, RegisterTaskDefinitionRequest.class);
-    } catch (IOException e) {
-      throw new UncheckedIOException(e);
+    if (!description.isEvaluateTaskDefinitionArtifactExpressions()) {
+      File artifactFile =
+          downloadTaskDefinitionArtifact(description.getResolvedTaskDefinitionArtifact());
+      try {
+        requestTemplate = mapper.readValue(artifactFile, RegisterTaskDefinitionRequest.class);
+      } catch (IOException e) {
+        throw new UncheckedIOException(e);
+      }
+    } else {
+      requestTemplate = description.getSpelProcessedTaskDefinitionArtifact();
     }
```

Comment:
Would you be able to point to a different spot where we have split logic like this for processing spel logic? I'm a bit hesitant in having more forked logic. For instance, what if enable SPEL processing is true but the artifact doesn't actually have any expressions (in that case, the logic btwn true and false would have to function the same). 

Is there a way we can share the logic instead of having forked logic?

Reply:
Your concern is valid, but no where in the code we have this situation where in we need to either download server group info from somewhere else or get it from pipeline JSON. In this particular scenario we have that scenario due to which we need to have this forked logic. And it is hard to figure out if SPEL processing is true but the artifact doesn't actually have any expressions.

Reply:
In either case, could we have the same logic apply across both scenarios, and merge the artifact with the processed artifact (for instance, have orca provide the processed artifact if it uses spel expressions)?